tests/ui-fulldeps/obtain-borrowck.rs

```diff
@@ -1,3 +1,10 @@
+//@ edition: 2021
+//@ run-pass
+//@ check-run-results
+//@ run-flags: --sysroot {{sysroot-base}} --edition=2021 {{src-base}}/auxiliary/obtain-borrowck-input.rs
+//@ ignore-stage1 (requires matching sysroot built with in-tree compiler)
```

Comment:
Incidentally, I was able to track down why these tests require stage 2 in ui-fulldeps.

It's because #110478 made "stage 1" tests in ui-fulldeps actually use the `stage 0` compiler, so the test tries to use an in-tree compiler (via API) with a bootstrap sysroot, and predictably fails.

I have an idea of how to solve this (pass along two sysroots and let fulldeps tests select the correct one), but I haven't decided whether it's worth the extra complexity.

Reply:
Oh god.

Reply:
@Zalathar do you mind opening an issue for that so we don't lose track of it (since you probably have more context than me) and other people might chime in?

Reply:
I haven't surveyed all of the `ui-fulldeps` tests, but I would be very surprised if any of the tests actually relies on having the *stage 2* sysroot being available?

Reply:
From what I can tell, ui-fulldeps treats fall into two categories:
- Those that use rustc crates in a way that doesn’t need a sysroot, so using stage 0 is fine and useful.
- Those that use rustc crates to invoke the compiler and this need a suitable sysroot, so they’re all ignore-stage1.

Reply:
Ah I see, thanks for the explanation.

Reply:
i would also caution you, like, weigh the time it would take to get these working with stage2 against the time it would take to just build both sysroots
it's not super common to need to run `ui-fulldeps` tests locally

Reply:
~~you don't want to know how long it took me to get #110478 working~~